### PR TITLE
update the amd ci pipeline

### PR DESCRIPTION
**Pull request by PeixuanZuo**
**Description**: Describe your changes.
AMD CI Test often failed because of AMD machine instability(deployed six agent), need manually rerun.
1.Lower one of the CI Test threshold to reduce CI Failed.
2.Add task retryCountOnTaskFailure to automatic retry CI Test task immediately.
3.Add get machine environment task to minotor the using of machine.

[Requires]
retryCountOnTaskFailure is a new feature, requires agent version 2.194.0 or later. Need to update the AMD_POOL agents.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml [hunk 1]
```diff
@@ -33,90 +33,110 @@ jobs:
...
+          --per_device_train_batch_size 8 \
+          --fp16 \
+          --dataloader_num_workers 1 \
+          --ort \
+          --skip_memory_metrics
+        python ../../orttraining/tools/ci_test/compare_huggingface.py \
+          ci-pipeline-actual.json \
+          ../../orttraining/tools/ci_test/results/ci-mi100.huggingface.bert-large-rocm4.3.1.json
     displayName: 'Run Python Hugging-Face BERT-L test'
+    retryCountOnTaskFailure: 2
```
Comment (ytaous) [inline tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:110]: >2 [](http://example.com/codeflow?start=29&length=1)

should we reduce it to 1 ?

Comment (PeixuanZuo) [reply tools/ci_build/github/azure-pipelines/orttraining-pai-ci-pipeline.yml:110]: Yes, may be 1 time is enough, avoid repeated failure when the pr does have error

File: tools/ci_build/github/pai/pai_get_thread.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+echo "agent name final character : $finalCharacter"
+targetRender=$((finalCharacter+128))
+
+echo -e "\n ---- rocm-smi"
+rocm-smi
+
+echo -e "\n ---- rocm-smi --showpids"
+rocm-smi --showpids
+
+echo -e "\n ---- rocm-smi --showpidgpus"
```
Comment (ytaous) [inline tools/ci_build/github/pai/pai_get_thread.sh:15]: is this really useful?

Comment (PeixuanZuo) [reply tools/ci_build/github/pai/pai_get_thread.sh:15]: Yes, this is to record the CI environment, we can see if there are other threads occupying our target GPU, and we have used this to find the occupancy problems caused by PTE team.

**Comment by PeixuanZuo** [comment]
> not sure what's the process, but might be good to coordinate with PTE/BIC team, as they are using -35 daily
> 
> In reply to: [1056435750](https://github.com/microsoft/onnxruntime/pull/10723#issuecomment-1056435750)

We can only update the agents we are using by click "update agent", But i don't have the permission to update.

**Comment by ytaous** [comment]
I have access, but let me check with AMD VM owner first tomorrow, and then I'll coordinate it with BIC team.

---
In reply to: [1056459692](https://github.com/microsoft/onnxruntime/pull/10723#issuecomment-1056459692) [](http://example.com/codeflow?ancestors=1056459692,1056459692)

File: orttraining/tools/ci_test/results/ci-mi100.huggingface.gpt2-rocm4.3.1.json [hunk 3]
```diff
@@ -53,5 +53,5 @@
             "loss": 1.5503
         }
     ],
-    "samples_per_second": 25.587
+    "samples_per_second": 24.756
```
Comment (pengwa) [inline orttraining/tools/ci_test/results/ci-mi100.huggingface.gpt2-rocm4.3.1.json:56]: >24.756 [](http://example.com/codeflow?start=26&length=6)

intresting, how do you choose this magic number?

**Review by pengwa — DISMISSED** [review]
LGTM. Thanks Peixuan for the fix! :) BTW, why the PR is a draft?

File: orttraining/tools/ci_test/results/ci-mi100.huggingface.gpt2-rocm4.3.1.json (hunk 3 above)
Comment (PeixuanZuo) [reply orttraining/tools/ci_test/results/ci-mi100.huggingface.gpt2-rocm4.3.1.json:56]: I did some experiments and compute a average number of them. If there is a better way I can try again.

**Comment by PeixuanZuo** [comment]
> LGTM. Thanks Peixuan for the fix! :) BTW, why the PR is a draft?

Because the agents in agent pool need update, I cann't test the retry function before it update.

**Review by ytaous — APPROVED** [review]
agent updated to 2.2

**Review by ytaous — APPROVED** [review]

**Pull request merged**